### ISSUE TITLE
How to remove HTML from "article document title"  before submitting to Apple News?

### ISSUE THREAD

**Issue by newvision2112** [alleyinteractive/apple-news#1191]
### Description of the bug

We are submitting our draft articles to Apple News. The issue is our post titles need to italicize names of magazines and newspapers so we have `<I></I>` and `<em></em>` tags in the title text. This works for WordPress showing the title on our website. 
But, it turns out Apple News will not accept "article document title" with HTML in them. This is not the title of the article that shows in Apple News. 

How do I configure the plugin strip the HTML characters from the "article document title" before submitting the JSON to Apple News?

**Comment by mogmarsh** [comment]
@newvision2112 you should be able to use the [apple_news_exporter_title](https://github.com/alleyinteractive/apple-news/wiki/Filter:-apple_news_exporter_title) filter

**Comment by newvision2112** [comment]
@mogmarsh Thank you. So when I remove HTML tags on that:

```
$title_start = $post->post_title; 
$title_start_2 = strip_tags($title_start); 
$title = apply_filters( 'apple_news_exporter_title', $title_start_2, $post->ID );
```

It removes the html from ALL titles in the JSON. So it looks like we lose the ability to have italics tags in live article headline that people read in Apple News. 

The article title JSON:
<img width="561" alt="SCR-20241106-kpja" src="https://github.com/user-attachments/assets/53ff18b5-de81-4284-92ee-57b7377208a5">

And the main headline:
<img width="584" alt="SCR-20241106-kqje" src="https://github.com/user-attachments/assets/e6bfdaea-8708-42b1-a86b-54c48ab00832">

Is there a way to only have the HTML removed from the "article document title"? But leave it in the regular title that shows up with the story on Apple News?

**Comment by mogmarsh** [comment]
@newvision2112 The [apple_news_generate_json](https://github.com/alleyinteractive/apple-news/wiki/Filter:-apple_news_generate_json) filter should work. It gives you an array representation of the json, so you would just change `$json['title']`

**Comment by mogmarsh** [comment]
@newvision2112 also, the tags should already be stripped: https://github.com/alleyinteractive/apple-news/blob/6eedc7922db8dac44a4191816a18a3b152492a8f/includes/apple-exporter/class-exporter.php#L212

**Comment by newvision2112** [comment]
@mogmarsh Thank you. I will work on apple_news_generate_json filter. 

I saw that code in the plugin `'title'      => wp_strip_all_tags( $this->content_title() ), `
but it did not remove the html tags `<I>title text</I>`  or `<em>title text</em>` from any of our JSON title fields in our submissions to Apple News drafts. 

Apple News flagged the problem to us and halted our site going live until we fix it.

**Comment by kevinfodness** [comment]
You're right, this is a bug. I was able to reproduce this on a vanilla installation of the plugin. It appears that the HTML in the title is getting partially transformed which is interfering with strip tags, which results in this output:

```json
"title": "Test &lt;i>italic title&lt;\/i>"
```

**Comment by newvision2112** [comment]
Thank you Kevin! Please let me know if I can test anything, or what a timeline for the fix might be. 
All best,
Dean

**Comment by mogmarsh** [comment]
@newvision2112 My [suggestion above](https://github.com/alleyinteractive/apple-news/issues/1191#issuecomment-2460539491) should work until we get the bug fixed. That filter runs at the very last minute and allows you to change almost anything.

**Comment by newvision2112** [comment]
@mogmarsh Thank you, but I don't understand the code enough to use apple_news_generate_json.

For now, to get Apple News to approve us, I used the code above to [strip all html from all titles](https://github.com/alleyinteractive/apple-news/issues/1191#issuecomment-2460525698)

**Comment by newvision2112** [comment]
Hello @mogmarsh  and @kevinfodness, just checking in on this. 
Thanks for your help, Dean

**Comment by kevinfodness** [comment]
I'd like to roll up a fix for this into the next version of the plugin, but it won't release until January, most likely.

**Comment by newvision2112** [comment]
@kevinfodness Thank you, appreciate it. I'll keep an eye out. --dean
